### DISCUSSION TITLE
Error: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?

### DISCUSSION THREAD

**Discussion by VS009** [concourse/concourse#8555] [Help & Support]
I am using the official docker base image,
>`image_resource:`
> `        type: registry-image`
> `        source:`
> `          repository: docker`

which allows me to use the docker toolkit in my builds (DinD). The task in which the image is used, then runs a build_docker.sh script,

> `run:`
>  `       path: sh`
>`         args: ["src/scripts/build_docker.sh"]`

which is supposed to run a few docker commands (namely `docker build` and `docker image`). However, I keep receiving this error when trying to run my build job.
I have so far also tried:
- mounting the socket of the host environment using` /var/run/docker.sock:/var/run/docker.sock` (DooD)
- running docker with `--privileged `flag
- `sudo dockerd `
- `sudo service docker start `

Each of these methods have incurred the same error where the Docker Daemon can not be connected to. 
Would appreciate any suggestions!

**Comment by marco-m** [comment]
if you final goal is simply to build a Docker image, you can save yourself a lot of pain by following https://concourse-ci.org/container-image-guides.html :-)
